### PR TITLE
Fix cmake/compile errors related to disabling xnnpack

### PR DESCRIPTION
**Pull request by rjknight**
- pytorch/pull/35607 attempted to fix the CMAKE errors described in
  issue #34606 unfortunately it did not allow xnnpack
  to be disabled if other nnpack options were eneabled. This PR
  fixes the compile problem introduced in PR35607

### CODE REVIEW

**Review by malfet — APPROVED** [review]

File: caffe2/CMakeLists.txt [hunk 1]
```diff
@@ -48,16 +48,23 @@ if(INTERN_BUILD_ATEN_OPS)
   list(APPEND Caffe2_DEPENDENCY_INCLUDE ${ATen_THIRD_PARTY_INCLUDE})
 endif()
 
+# {Q/X,etc} NPACK support is enabled by default, if none of these options
+# are selected, turn this flag ON to incidate the support is disabled
+set(NNPACK_AND_FAMILY_DISABLED OFF)
```
Comment (xuhdev) [inline caffe2/CMakeLists.txt:53]: Could you reuse the variable at https://github.com/pytorch/pytorch/blob/379e717a1b35817c7009c3ca717fe53dabab5510/cmake/Dependencies.cmake#L196 ?

File: caffe2/CMakeLists.txt [hunk 2]
```diff
@@ -48,16 +48,23 @@ if(INTERN_BUILD_ATEN_OPS)
   list(APPEND Caffe2_DEPENDENCY_INCLUDE ${ATen_THIRD_PARTY_INCLUDE})
 endif()
 
+# {Q/X,etc} NPACK support is enabled by default, if none of these options
+# are selected, turn this flag ON to incidate the support is disabled
```
Comment (xuhdev) [inline caffe2/CMakeLists.txt:52]:
```suggestion
# are selected, turn this flag ON to indicate the support is disabled
```

**Comment by rgommers** [comment]
This LGTM. It fixes a very annoying problem that quite a few people are running into, and CI is green. The workaround increases build times significantly. 

@xuhdev could this be landed as is, and the two open comments (which seem non-blocking) addressed in a follow-up PR?

**Comment by xuhdev** [comment]
@rgommers No problem

**Pull request closed without merging**